web/custom-plugins/forecast-of-response/src/components/Dashboard.tsx

```diff
@@ -1,14 +1,98 @@
-import React from 'react'
-import { DmtSettings } from '../Types'
-import AccessControlList from "./AccessControl/AccessControlList"
+import React, { useContext, useEffect, useState } from 'react'
+import { MapContainer, Marker, Popup, TileLayer, Tooltip } from "react-leaflet"
+import 'leaflet/dist/leaflet.css'
+import L from 'leaflet'
+import styled from "styled-components"
+import { CompactCommentView } from "./Comments"
+import useSearch from "../hooks/useSearch"
+import { Blueprints } from "../Enums"
+import { TComment, TOperation } from "../Types"
+import { AuthContext, DmssAPI } from '@dmt/common'
+
+delete L.Icon.Default.prototype._getIconUrl
+
+L.Icon.Default.mergeOptions({
+  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
+  iconUrl: require('leaflet/dist/images/marker-icon.png'),
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+})
+
+const CardWrapper = styled.div`
+  border: darkgrey 1px solid;
+  border-radius: 4px;
+  display: flex;
+  flex-direction: column;
+  box-shadow: darkgrey 0 2px 8px 2px ;
+`
+
+const StyledMapContainer = styled(MapContainer)`
+  height: 100%;
+  width: 100%;
+  border-top: darkgrey 1px solid;
+`
+type CoordinateTuple = [number, number, string]
+
+function calculateUTMCenter(coordinates: CoordinateTuple[]|undefined): [number, number] {
+  if(!coordinates) return [57.4, 4.4]  // Default value somewhere in the Northsea
+  const xSum = coordinates.reduce((accum, b) => accum + b[0], 0)
+  const ySum = coordinates.reduce((accum, b) => accum + b[1], 0)
+  return [xSum / coordinates.length, ySum / coordinates.length]
+}
+
+const Dashboard = (): JSX.Element => {
+  const { token } = useContext(AuthContext)
+  const dmssAPI = new DmssAPI(token)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [coordinates, setCoordinates] = useState<CoordinateTuple[]>()
+  const [comments, commentsLoading] = useSearch(Blueprints.Comment)
+  const [operations] = useSearch(Blueprints.OPERATION)
+
+  useEffect(()=>{
+    setLoading(true)
+    if(!operations) return
+    Promise.all(operations.map((operation: TOperation): CoordinateTuple =>{
+      return dmssAPI
+      .getDocumentById({ dataSourceId: "ForecastDS", documentId: operation.location._id})
+      .then((document): any => {
+        const location = document.document
+        return [location.lat, location.long, operation.name]
+      })
+  })).then((coordinates: CoordinateTuple[])=> {
+      setCoordinates(coordinates)
+      setLoading(false)
+    })
+  },[operations])
+
+  if(loading) return <>Loading...</>
 
-const Dashboard = (props: DmtSettings): JSX.Element => {
-  const { settings } = props
   return (
-    <>
-      <i>My dashboard</i>
-      <AccessControlList/>
-    </>
+    <div style={{ display: "flex", minHeight: "500px", maxHeight: "900px"}}>
+      <CardWrapper style={{ width: "70%" }}>
+        <h3 style={{ margin: "5px" }}>Ongoing operations (3)</h3>
```

Comment:
can length of the operations variable be used instead of 3?

Reply:
fixed!